pkg/utils/errors/errors.go

```diff
@@ -0,0 +1,112 @@
+package errors
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+)
+
+// repoRoot is used to limit frame processing to only those created by this project.  See frameUnstack() for why it is hardcoded
+const repoRoot = "containerized-data-importer"
+
+// prettyError wraps a passed in message and the internally generated stack trace.  `message` may be
+// an string or an error interface.
+type prettyError struct {
+	message    string
+	traceStack *trace
+	frameStack []runtime.Frame
+}
+
+// Error satisfies the built-in error interface
+func (e prettyError) Error() string {
+	return fmt.Sprintf("ERROR: %v\n%s", e.message, e.traceStack)
+}
+
+// String satisfies the built-in Stringer interface
+func (e prettyError) String() string {
+	return e.Error()
+}
+
+// Err is intended to behave as the built-in errors.New() function.  Use this function to generate simple prettyErrors.
```

Comment:
Seems like `Err` and `Errf` should be symmetric in their description. One behaves like errors.New and the other like fmt.Errorf, yet both call `newError`

Reply:
This is actually more symmetric with the built ins that it appears.  `fmt.Errorf()` is just
```
func Errorf(format string, a ...interface{}) error {
	return errors.New(Sprintf(format, a...))
}
```

Reply:
ack